apps/walletshield/main.go

```diff
@@ -176,7 +213,31 @@ func (s *Server) SendTestProbes(d time.Duration, testProbeCount int) {
 	for {
 		packetsTransmitted++
 		t := time.Now()
-		_, err := s.session.BlockingSendUnreliableMessage(s.target.Name, s.target.Provider, blob)
+
+		// Select a target service node and compute the DestinationIdHash
+
+		doc := s.thin.PKIDocument()
+		if doc == nil {
+			s.log.Errorf("Failed to retrieve PKI document")
+			return
+		}
+		var destinationIdHash [32]byte
+		if len(doc.ServiceNodes) > 0 {
+			node := doc.ServiceNodes[0] // Choose the first service node for simplicity
+			destinationIdHash = hash.Sum256(node.IdentityKey)
+		} else {
+			s.log.Errorf("No service nodes available in PKI document")
+			return
+		}
+		surbID := &[sConstants.SURBIDLength]byte{}
+		_, err := rand.Reader.Read(surbID[:])
+		if err != nil {
+			panic(err)
+		}
+		recipientQueueID := []byte("http_proxy")
+
+		timeoutCtx, _ := context.WithTimeout(context.TODO(), (time.Second * 200))
+		_, err = s.thin.BlockingSendMessage(timeoutCtx, blob, &destinationIdHash, recipientQueueID)
```

Comment:
This could be more D.R.Y.

Ideally test probe feature very closely matches "real" request.